bin/reth/src/args/miner_args.rs

```diff
@@ -0,0 +1,78 @@
+use clap::Args;
+use reth_primitives::Address;
+
+/// Parameters for configuring the miner
+#[derive(Debug, Args, PartialEq, Default)]
+pub struct MinerArgs {
+    /// Enable mining
+    #[arg(long = "mine", help_heading = "Miner")]
+    pub mine: bool,
+
+    /// Public address for block mining rewards
+    #[arg(long = "miner.etherbase", help_heading = "Miner")]
+    pub etherbase: Option<Address>,
+
+    /// Block extra data set by the miner
+    #[arg(long = "miner.extradata", help_heading = "Miner")]
+    pub extradata: Option<String>,
+
+    /// Target gas ceiling for mined blocks
+    #[arg(long = "miner.gaslimit", help_heading = "Miner")]
+    pub gaslimit: Option<u64>,
+
+    /// Minimum gas price for mining a transaction
+    #[arg(long = "miner.gasprice", help_heading = "Miner")]
+    pub gasprice: Option<u64>,
+
+    /// Comma separated HTTP URL list to notify of new work packages
+    #[arg(long = "miner.notify", help_heading = "Miner")]
+    pub notify: Option<String>,
+
+    /// Comma separated HTTP URL list to notify of new work packages
+    #[arg(long = "miner.notify.full", help_heading = "Miner")]
+    pub notify_full: bool,
+
+    /// Disable remote sealing verification
+    #[arg(long = "miner.noverify", help_heading = "Miner")]
+    pub noverify: bool,
+
+    /// Time interval to recreate the block being mined
+    #[arg(long = "miner.recommit", help_heading = "Miner")]
+    pub recommit: Option<u32>,
+
+    /// Number of CPU threads to use for mining
+    #[arg(long = "threads", help_heading = "Miner")]
+    pub num_threads: Option<u32>,
```

Comment:
we need to enforce some bounds here.
I'd like to use usize here also

requires num_cpu crate, this should work:

```rust
 value_parser = RangedU64ValueParser::<usize>::new().range(1..num_cpus::get()),
```